vividus-extension-selenium/src/main/java/org/vividus/monitor/AbstractPublishingAttachmentOnFailureMonitor.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.monitor;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+
+import org.jbehave.core.model.Scenario;
+import org.jbehave.core.steps.NullStepMonitor;
+import org.vividus.context.RunContext;
+import org.vividus.model.RunningScenario;
+import org.vividus.model.RunningStory;
+import org.vividus.reporter.event.AttachmentPublishEvent;
+import org.vividus.reporter.model.Attachment;
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.softassert.event.AssertionFailedEvent;
+
+public abstract class AbstractPublishingAttachmentOnFailureMonitor extends NullStepMonitor
+{
+    private final ThreadLocal<Boolean> publishAttachmentOnFailureEnabled = ThreadLocal.withInitial(() -> Boolean.FALSE);
+
+    private final String metaNameToSkipPublishing;
+    private final RunContext runContext;
+    private final EventBus eventBus;
+    private final IWebDriverProvider webDriverProvider;
+
+    public AbstractPublishingAttachmentOnFailureMonitor(RunContext runContext, IWebDriverProvider webDriverProvider,
+            EventBus eventBus, String metaNameToSkipPublishing)
+    {
+        this.runContext = runContext;
+        this.webDriverProvider = webDriverProvider;
+        this.eventBus = eventBus;
+        this.metaNameToSkipPublishing = metaNameToSkipPublishing;
+    }
```

Comment:
keep the same order for fields, constructor parameters and fields initializations in the constructor